src/lib/index.js

```diff
@@ -350,19 +350,25 @@ lib.noneOrAll = function(containerIn, containerOut, attrList) {
     }
 };
 
-/** merge data array into calcdata items
+lib.mergeArray = function(traceAttr, cd, cdAttr) {
+    if(Array.isArray(traceAttr)) {
+        var imax = Math.min(traceAttr.length, cd.length);
+        for(var i = 0; i < imax; i++) cd[i][cdAttr] = traceAttr[i];
+    }
+};
+
+/** fill data array into calcdata items
  *
  * @param {array} traceAttr : trace attribute
  * @param {object} cd : calcdata trace
  * @param {string} cdAttr : calcdata key
  * @param {function} [fn] : optional function to apply to each array item
  */
-lib.mergeArray = function(traceAttr, cd, cdAttr, fn) {
+lib.fillArray = function(traceAttr, cd, cdAttr, fn) {
     fn = fn || lib.identity;
```

Comment:
That won't make that much of a difference (at least in Chrome 58)

![image](https://user-images.githubusercontent.com/6675409/26840662-d73f6802-4ab4-11e7-9df0-9ea615163126.png)


https://gist.github.com/etpinard/4a5c7983a619efb4c98cb927529b9af6